cmd/test_stores/main.go

```diff
@@ -408,15 +508,26 @@ func (fri *FakeRefineryInstance) runProcessor(opts CmdLineOptions, nodeIndex int
 			fmt.Printf("processor %d: got trace %s\n", nodeIndex, tid)
 			traceIDs.Add(tid)
 		case <-statusTicker.C:
+			ctx, root := fri.tracer.Start(context.Background(), "processor")
+			addSpanFields(root, map[string]interface{}{
+				"nodeIndex":     nodeIndex,
+				"interval":      interval,
+				"num_trace_ids": len(traceIDs.Members()),
+			})
 			tids := traceIDs.Members()
 			if len(tids) == 0 {
 				continue
 			}
+			_, span := fri.tracer.Start(ctx, "get_status_for_traces")
+			addSpanField(span, "num_trace_ids", len(tids))
 			statuses, err := fri.store.GetStatusForTraces(tids)
 			if err != nil {
-				fmt.Println(err)
+				fri.addException(span, err)
+				span.End()
 				continue
 			}
+			span.End()
```

Comment:
Using `defer span.End()` would save you from having to call end in both cases here

Reply:
same as above -- that wouldn't work here because the function is in an infinite loop.